catalog/app/utils/json-schema.js

```diff
@@ -0,0 +1,84 @@
+import * as R from 'ramda'
+import isArray from 'lodash/isArray'
+import isNumber from 'lodash/isNumber'
+import isObject from 'lodash/isObject'
+import isString from 'lodash/isString'
+import isUndefined from 'lodash/isUndefined'
+
+function isSchemaArray(optSchema) {
+  return R.prop('type', optSchema) === 'array'
+}
+
+function isSchemaObject(optSchema) {
+  return R.prop('type', optSchema) === 'object'
+}
+
+function isSchemaString(optSchema) {
+  return R.prop('type', optSchema) === 'string'
+}
+
+function isSchemaNumber(optSchema) {
+  return R.prop('type', optSchema) === 'number'
+}
+
+function isSchemaBoolean(optSchema) {
+  return R.prop('type', optSchema) === 'boolean'
+}
+
+function isSchemaNull(optSchema) {
+  return R.prop('type', optSchema) === 'null'
+}
+
+export function isSchemaEnum(optSchema) {
```

Comment:
it's a little strange that only some of these test functions are exported

Reply:
I export functions that are used outside. If I export every function here it would be harder to refactor and maintain, because I should find if this function is used somewhere else.